src/imitation/scripts/train_imitation.py

```diff
@@ -3,173 +3,139 @@
 import logging
 import os.path as osp
 import pathlib
-import warnings
-from typing import Any, Mapping, Optional, Sequence, Type, cast
+from typing import Any, Dict, Mapping, Optional, Sequence, cast
 
 import numpy as np
 from sacred.observers import FileStorageObserver
-from stable_baselines3.common import policies, utils, vec_env
 
-from imitation.algorithms import bc as bc_algorithm
 from imitation.algorithms.dagger import SimpleDAggerTrainer
 from imitation.data import rollout, types
 from imitation.scripts.config.train_imitation import train_imitation_ex
+from imitation.scripts.ingredients import bc as bc_ingredient
 from imitation.scripts.ingredients import demonstrations, environment, expert
 from imitation.scripts.ingredients import logging as logging_ingredient
-from imitation.scripts.ingredients import train
+from imitation.scripts.ingredients import policy_evaluation
 
 logger = logging.getLogger(__name__)
 
 
-@train_imitation_ex.capture(prefix="train")
-def make_policy(
-    venv: vec_env.VecEnv,
-    policy_cls: Type[policies.BasePolicy],
-    policy_kwargs: Mapping[str, Any],
-    agent_path: Optional[str],
-) -> policies.BasePolicy:
-    """Makes policy.
-
-    Args:
-        venv: Vectorized environment we will be imitating demos from.
-        policy_cls: Type of a Stable Baselines3 policy architecture.
-            Specify only if policy_path is not specified.
-        policy_kwargs: Keyword arguments for policy constructor.
-            Specify only if policy_path is not specified.
-        agent_path: Path to serialized policy. If provided, then load the
-            policy from this path. Otherwise, make a new policy.
-            Specify only if policy_cls and policy_kwargs are not specified.
-
-    Returns:
-        A Stable Baselines3 policy.
-    """
-    policy_kwargs = dict(policy_kwargs)
-    if issubclass(policy_cls, policies.ActorCriticPolicy):
-        policy_kwargs.update(
-            {
-                "observation_space": venv.observation_space,
-                "action_space": venv.action_space,
-                # parameter mandatory for ActorCriticPolicy, but not used by BC
-                "lr_schedule": utils.get_schedule_fn(1),
-            },
-        )
-    policy: policies.BasePolicy
-    if agent_path is not None:
-        warnings.warn(
-            "When agent_path is specified, policy_cls and policy_kwargs are ignored.",
-            RuntimeWarning,
+def _add_expert_stats_to_stats_dict_if_all_are_trajectories_with_reward(
```

Comment:
This seems like an unreasonably large function name, but not sure I can think of a way to make it shorter.

Reply:
I refactored this a bit to hopefully make it easier to read. What do you think about it?